pkg/daemon/ceph/client/pool.go

```diff
@@ -400,26 +412,50 @@ func createTwoStepCrushRule(context *clusterd.Context, clusterInfo *ClusterInfo,
 		return errors.Errorf("failure and subfailure domains cannot be identical, current is %q", pool.FailureDomain)
 	}
 
+	crushMap, err := getCurrentCrushMap(context, clusterInfo)
```

Comment:
These changes look very similar to the implementation in `createHybridCrushRule()`. Could it be factored to a common helper?